src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -372,4 +372,15 @@ private void setAssignedTeamIdForExerciseAndUser(Exercise exercise, User user) {
             });
         }
     }
+
+    /**
+     * Sets the transient attribute "testRepositoryUrl" if the exercises is a programming exercise
+     *
+     * @param exercise the exercise for which to set the url
+     */
+    public void setTestRepositoryUrlForProgrammingExercise(Exercise exercise) {
+        if (exercise instanceof ProgrammingExercise) {
+            exercise.setTestRepositoryUrlTransient(programmingExerciseService.findByIdWithEagerStudentParticipations(exercise.getId()).getTestRepositoryUrl());
```

Comment:
Why do you fetch again? I think you can simply cast the java object here

Reply:
Yes, you are right, thank you for the tip